fifteen_min/src/find_amenities.rs

```diff
@@ -0,0 +1,135 @@
+use map_gui::tools::{ChooseSomething, ColorLegend};
+use map_gui::ID;
+use map_model::AmenityType;
+use widgetry::{
+    Cached, Choice, Color, Drawable, EventCtx, GfxCtx, HorizontalAlignment, Line, Panel,
+    SimpleState, State, TextExt, Transition, VerticalAlignment, Widget,
+};
+
+use crate::isochrone::{Isochrone, Options};
+use crate::viewer::{draw_star, HoverKey, HoverOnBuilding};
+use crate::App;
+
+/// Calculate isochrones around each amenity on a map and merge them together using the min value
+pub struct FindAmenity {
+    options: Options,
+}
+
+impl FindAmenity {
+    pub fn new(ctx: &mut EventCtx, options: Options) -> Box<dyn State<App>> {
+        ChooseSomething::new(
+            ctx,
+            "Choose an amenity",
+            AmenityType::all()
+                .into_iter()
+                .map(|at| Choice::new(at.to_string(), at))
+                .collect(),
+            Box::new(move |at, ctx, app| {
+                let multi_isochrone = create_multi_isochrone(ctx, app, at, options.clone());
+                return Transition::Push(Results::new(ctx, app, multi_isochrone, at));
```

Comment:
Hmm, when I leave the results screen by pressing escape, I go back to the menu. From a UX perspective, I would expect to return to the main viewer. That's happening because the stack winds up being [viewer, ChooseSomething, Results]. I'd vote for changing this to `Transition::Replace`

Reply:
Ah I can see that, I think I was expecting the menu because I wanted to go back and look at a different amenity, that might be more for testing, I can see the argument that a typical user wants to see one amenity and once they are done go back to the viewer.

Reply:
I could see that too. I'm fine either way, it's up to you. When we eventually think about UX design properly, we can answer this better, knowing what use cases people will have here

Reply:
Sounds good to me, I changed to replace for now and we can always circle back.